text/java/0000-composite-draft-branch.md

```diff
@@ -0,0 +1,39 @@
+# Create a draft branch for composite/meta buildpacks
+
+## Summary
+
+Currently component buildpack bumps are created as PRs against the main branch, e.g. see [here](https://github.com/paketo-buildpacks/java/pull/873).
+This RFC proposes changing the CI to create these dependency PRs against a new branch, e.g. `draft`.    
+
+## Motivation
+
+This will allow us to achieve the following improvements:
+
+1. Auto-merging of component buildpack PRs into this draft branch 
+2. Manual review can be performed with a single PR before merging to main
+2. Integration tests can be run on merging to main ensuring all changes in the release are covered
+
+## Detailed Explanation
+
+Currently the component dependency PRs must be manually approved and merged separately, even though their changes do not cause merge conflicts. Auto-merging these will reduce the maintainence burden involved here.
+Integration tests can configured to run on merges to the main branch, which could happen once per week before a release, or more frequently if necessary for a specific purpose.
+
+## Rationale and Alternatives
+
+1. Do not change the current CI/branch configuration - integration tests could be run via a manual step before release.
+2. Do not change the current CI/branch configuration - create new release tags as a separate initial step before a release is published. Integration tests could be run on tag creation instead, with the publish action potentially triggered from passed tests. 
```

Comment:
Should we maybe add - and of course evaluate - using [Renovate](https://docs.renovatebot.com) as an alternative or additional means to achieve our goals here?

1. There is a nice [Dependency Dashboard](https://docs.renovatebot.com/key-concepts/dashboard/), showing a repo's dependency and deferred updates and allows to control which update PRs are done when (see [Dependency Dashboard Approval workflow](https://docs.renovatebot.com/key-concepts/dashboard/#dependency-dashboard-approval-workflow).
2. You can e.g. [group update PRs](https://docs.renovatebot.com/key-concepts/pull-requests/#grouped-updates-with-different-versions) to reduce the number of individual PRs. Applying all patch updates in a batch PR is an easy option.
3. Renovate can add labels, so we could control to what extend CI will execute tests depending on the updates we do.

The "easy to use" might hit it's limit when we look at updating dependencies like Tomcat, because there is a limited number of [datasources](https://docs.renovatebot.com/modules/datasource/) Renovate supports. On the other hand side, [Tomcat seems to at least have proper version tags](https://github.com/apache/tomcat/tags).

TL;DR We should investigate how far we can lower the CI resources used for testing and review effort required merging with using Renovate. If this evaluation runs successfully, it might be worth rolling this out to other Paketo repositories via a separate, project scoped RFC.

cc @pbusko

Reply:
@loewenstein I think renovate could be a useful tool to explore. If we are able to get all the paketo buildpacks using the same tooling/workflows, then refactoring to use a tool like Renovate would impact all buildpacks equally.

Without digging into whether renovate can solve this particular problem in this RFC - bumping component buildpack versions in the meta buildpack - I think a good first step would be to unify the workflows of the buildpacks.

Also, the way I read this RFC, it doesn't try to address how dependencies (e.g. Tomact) are bumped within the component buildpacks.

Reply:
@robdimsdale well, the RFC sounds like there is effort involved changing CI and while maybe not too much still significant. So I thought rather than aligning first and then changing we could just as well explore the alternative and align in that - if it turns out to be the superior choice.

Vacation season has started over here, but we have a bit of experience with Renovate, so we could probably provide some insights and spend some cycles, if this is considered useful.

Reply:
I would like to explore Renovate. I'm not happy with Dependabot, and it not updating transitive dependencies. I would also like to see if we could get Renovate to replace our CI jobs that run to update composite buildpacks (i.e. teach it to know about dependent buildpacks and submit PRs for those as well).

My $0.02 is that it should be something separate we explore though, to not hold up the changes here. Also as @robdimsdale mentioned, it would really be project-wide. I think a good first step would be to do an exploration, possibly in the [explorations repo](https://github.com/paketo-community/explorations) or maybe as a separate repo since we will probably need to adjust CI & integrate with the Renovate services.

Another concern with Renovate is whether can we get a managed service of it that provides all the functionality we require, or if there is something we need to host. If it's the latter, we have to be very careful with the amount of maintenance that requires.

Reply:
> I would also like to see if we could get Renovate to replace our CI jobs that run to update composite buildpacks (i.e. teach it to know about dependent buildpacks and submit PRs for those as well).

We're using Renovate for updating composite buildpack dependencies, as well as builder dependencies, like lifecycle version, stack versions and included buildpacks with the [regex manager](https://docs.renovatebot.com/modules/manager/regex/) feature + [github releases datasource](https://docs.renovatebot.com/modules/datasource/#github-releases-datasource).

> Another concern with Renovate is whether can we get a managed service of it that provides all the functionality we require, or if there is something we need to host. If it's the latter, we have to be very careful with the amount of maintenance that requires.

In a nutshell, renovate is just a CLI, which makes it extremely easy to self-host (github action (official one is available)/k8s job/any runner capable of calling renovate CLI). Also there's a [free GitHub App](https://github.com/apps/renovate) ready for use. Functionality-wise all options are equal.

Reply:
I would like to propose that we move the Renovate topic to a new RFC. It really is something that should be Paketo-wide and this is a Java-specific RFC so I don't think it has the visibility necessary. Also, we have [specific needs for this RFC](https://github.com/paketo-buildpacks/java/pull/828) which we don't want to block by a larger discussion on Renovate.

If everyone is OK with this, please 👍 this comment. Thanks